sparse-index.c

```diff
@@ -574,6 +603,11 @@ static int clear_skip_worktree_from_present_files_sparse(struct index_state *ist
 	int path_count = 0;
 	int to_restart = 0;
 
+	if (!core_apply_sparse_checkout ||
+	    core_virtualfilesystem ||
+	    sparse_expect_files_outside_of_patterns)
+		return to_restart;
+
```

Comment:
I think this `if` (and the fact that `microsoft/git` needs `core_virtualfilesystem` checks here) belongs in `clear_skip_worktree_from_present_files()`, further down.

Reply:
Yes, absolutely! Good eyes.